src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder employeeSalaryOfPeriod = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+
+        int salary = 0;
+        LocalDate localDate;
+        LocalDate localDateFrom;
+        LocalDate localDateTo;
+        localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        localDateTo = LocalDate.parse(dateTo, FORMATTER);
+        for (int j = 0; j < names.length; j++) {
+            salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                String[][] dataInfo = new String[data.length][4];
+                dataInfo[i] = data[i].split(" ");
+                localDate = LocalDate.parse(dataInfo[i][0], FORMATTER);
+                String employeeNames = dataInfo[i][1];
+                int workingHours = Integer.parseInt(dataInfo[i][2]);
+                int salaryForHours = Integer.parseInt(dataInfo[i][3]);
```

Comment:
instead of using magic numbers let's create constants for indexes (for example SALARY_INDEX)